share/completions/history.fish

```diff
@@ -1,10 +1,33 @@
-complete -c history -r -l prefix --description "Match items starting with prefix"
-complete -c history -r -l contains --description "Match items containing string"
-complete -c history -l search -s s --description "Prints commands from history matching query"
-complete -c history -l delete -s d --description "Deletes commands from history matching query"
-complete -c history -l clear --description "Clears history file"
-complete -c history -l merge -s m --description "Incorporate history changes from other sessions"
-complete -c history -l exact -s e --description "Match items in the history that are identicial"
+set -g __fish_history_all_commands search delete save merge clear
```

Comment:
No need for a global if you expand it directly - use double-quotes where you use it here.